algorithms/src/snark/marlin/marlin.rs

```diff
@@ -504,10 +574,10 @@ where
             .collect();
 
         // Gather commitment randomness together.
-        let commitment_randomnesses: Vec<Randomness<E>> = circuit_proving_key
-            .circuit_commitment_randomness
+        let commitment_randomnesses: Vec<Randomness<E>> = keys_to_constraints
+            .keys()
+            .flat_map(|pk| pk.circuit_commitment_randomness.clone())
             .clone()
```

Comment:
is this clone needed, now that we clone in the `flat_map` above?

Reply:
Removed the outer clone!